### DISCUSSION TITLE
Too many requests

### DISCUSSION THREAD

**Discussion by Tihomir971** [supabase/supabase#10380] [Questions] [answered]
I'm using @supabase/auth-helpers-sveltekit and getSupabase(event) inside load function. Is it possible to eliminate reload on every event like switch browser window?

`export const load: PageLoad = async (event) => {
	const { session, supabaseClient } = await getSupabase(event);
	if (!session) {
		throw redirect(303, '/');
	}

	const { data: testTable } = await supabaseClient.from('test').select('*');
	return {
		testTable,
		user: session. User
	};
};`

**Comment by silentworks** [comment, marked as answer]
This is because each time you switch tabs an even is triggered by the supabase-js library, you can bypass this by checking the event being fired before allowing `invalidate` to run inside of your `+layout.svelte` file.

```js
onMount(() => {
  const {
	  data: { subscription }
  } = supabaseClient.auth.onAuthStateChange((event, _) => {
	  console.log({ event })
	  if (event !== 'SIGNED_IN') {
		  invalidate('supabase:auth')
	  }
  })

  return () => {
	  subscription.unsubscribe()
  }
})
```

**Comment by MaHDiaLaGaB** [comment]
hello,i have the same issue here simple request from my backend generate 450 request in Supabase do i have any problem here in my small code.
```
useEffect(() => {
    const {
      data: { subscription },
    } = supabase.auth.onAuthStateChange(() => {
      router.refresh();
    });

    return () => {
      subscription.unsubscribe();
    };
  }, [router, supabase]);
  ```
